src/ai/susi/server/api/aaa/PasswordChangeService.java

```diff
@@ -0,0 +1,116 @@
+/**
+ *  PasswordResetService
+ *  Copyright 6/7/17 by Dravit Lochan, @DravitLochan
+ *
+ *  This library is free software; you can redistribute it and/or
+ *  modify it under the terms of the GNU Lesser General Public
+ *  License as published by the Free Software Foundation; either
+ *  version 2.1 of the License, or (at your option) any later version.
+ *
+ *  This library is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ *  Lesser General Public License for more details.
+ *
+ *  You should have received a copy of the GNU Lesser General Public License
+ *  along with this program in the file lgpl21.txt
+ *  If not, see <http://www.gnu.org/licenses/>.
+ */
+package ai.susi.server.api.aaa;
+
+import ai.susi.DAO;
+import ai.susi.json.JsonObjectWithDefault;
+import ai.susi.server.*;
+import ai.susi.tools.TimeoutMatcher;
+import org.eclipse.jetty.util.log.Log;
+import org.json.JSONObject;
+
+
+import javax.servlet.http.HttpServletResponse;
+import java.util.regex.Pattern;
+
+/**
+ * Created by dravit on 6/7/17.
+ * test locally at http://127.0.0.1:4000/aaa/changepassword.json
+ * send the following three parameters :
+ * changepassword : contains email id of the user who's password has to be changed
+ * password : current password
+ * newpassword : new password
+ */
+public class PasswordChangeService extends AbstractAPIHandler implements APIHandler{
+    @Override
+    public String getAPIPath() {
+        return "/aaa/changepassword.json";
+    }
+
+    @Override
+    public BaseUserRole getMinimalBaseUserRole() {
+        return BaseUserRole.USER;
+    }
+
+    @Override
+    public JSONObject getDefaultPermissions(BaseUserRole baseUserRole) {
+        return null;
+    }
+
+    @Override
+    public ServiceResponse serviceImpl(Query post, HttpServletResponse response, Authorization rights, JsonObjectWithDefault permissions) throws APIException {
+        JSONObject result = new JSONObject();
+        result.put("accepted", false);
+
+        String useremail = post.get("changepassword", null);
+        String password = post.get("password", null);
+        String newpassword = post.get("newpassword",null);
```

Comment:
Sending passwords to the server as plaintext doesn't seem to be the right thing to do. Can you send a hash and then decrypt the hash on the server if needed, ideally that shouldn't be done ever. Storing hashes and checking for hashes seems to be more promising.

But I think that is however a much larger pull request than the one being looked at here.